app/controllers/admin/payments_controller.rb

```diff
@@ -1,13 +1,20 @@
 module Admin
   class PaymentsController < BaseAdminController
     before_action :ensure_service_operator
-    before_action :find_payroll_run, except: [:index]
-    before_action :find_payment, except: [:index]
+    before_action :find_payroll_run, except: [:index, :show]
+    before_action :find_payment, except: [:index, :show]
```

Comment:
non blocker but think i would opt to whitelist this, or inline as extraction doesn't really have much benefit